macosx/CMakeLists.txt

```diff
@@ -33,6 +36,12 @@ add_subdirectory(VDKQueue)
 
 add_executable(${TR_NAME}-mac MACOSX_BUNDLE)
 
+target_compile_options(${TR_NAME}-mac
+    PRIVATE
+        # equivalent of XCODE_ATTRIBUTE_SWIFT_OBJC_BRIDGING_HEADER "macosx/Transmission-Bridging-Header.h"
+        $<$<COMPILE_LANGUAGE:Swift>:-import-objc-header>
+        $<$<COMPILE_LANGUAGE:Swift>:${CMAKE_CURRENT_SOURCE_DIR}/Transmission-Bridging-Header.h>)
```

Comment:
The PR is called "Adopting Swiftlint" but these changes here and adding bridging headers are related to Swift in general, not SwiftLint, so let's rename the PR or move all these changes back to https://github.com/transmission/transmission/pull/4540.

Reply:
Well, I'm putting in this PR what doesn't break `make`. Any actual Swift file addition would strictly require `ninja` or `xcode` instead of `make`, hence how I'm doing the split. And the most important addition here is Swiftlint.